smctemp.cc

```diff
@@ -388,35 +401,68 @@ double SmcTemp::GetCpuTemp() {
     return temp;
   }
 #elif defined(ARCH_TYPE_ARM64)
-  std::vector<std::string> sensors{
-      static_cast<std::string>(kSensorTp01),
-      static_cast<std::string>(kSensorTp05),
-      static_cast<std::string>(kSensorTp0d),
-      static_cast<std::string>(kSensorTp0h),
-      static_cast<std::string>(kSensorTp0l),
-      static_cast<std::string>(kSensorTp0p),
-      static_cast<std::string>(kSensorTp0x),
-      static_cast<std::string>(kSensorTp0b),
-      static_cast<std::string>(kSensorTp09),
-      static_cast<std::string>(kSensorTp0t),
-  };
+  std::vector<std::string> sensors;
+  std::vector<std::string> aux_sensors;
+
+  if (getCPUModel() == "Apple M2") { // Apple M2
+    // CPU performance core 1 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp01));
+    // CPU performance core 2 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp09));
+    // CPU performance core 3 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp0f));
+    // CPU performance core 4 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp0n));
+    // CPU efficient core 1 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp05));
+    // CPU efficient core 2 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp0d));
+    // CPU efficient core 3 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp0j));
+    // CPU efficient core 4 temperature
+    sensors.emplace_back(static_cast<std::string>(kSensorTp0r));
+  } else { // Apple M1
```

Comment:
This looks like the M1 pro or M1 max, probably not a bad idea to else if with the CPU model and just error out if the model is not supported

Reply:
> probably not a bad idea to else if with the CPU model and just error out if the model is not supported

It seems reasonable for me. Could you modify code like below?
```console
if (getCPUModel() == "Apple M1") { // Apple M1
  ...
} else if (getCPUModel() == "Apple M2") { // Apple M2
  ...
} else { // Not supported
  // return 0.0 to indicate that the CPU is not supported
  return temp;
}
```

Reply:
Done in an equivalent way (to account for finer detection of the cpu model)